From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: RoccoDev <hey@rocco.dev>
Date: Mon, 26 Apr 2021 09:50:16 +0200
Subject: [PATCH] (KigPaper) Kick after receiving invalid creative action
 packet

Original code by ProjectKig, licensed under GPLv3
You can find the original code on https://github.com/ProjectKig/KigPaper

diff --git a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index 5b543a3f115fc80778db662d8557d12d3bd36942..5a6e05543ec3c8925a7d98b4e05272d1e26a9aac 100644
--- a/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/src/main/java/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -3461,6 +3461,8 @@ public class ServerGamePacketListenerImpl implements ServerPlayerConnection, Tic
                 this.dropSpamTickCount += 20;
                 this.player.drop(itemstack, true);
             }
+        } else {
+            disconnect("Not in creative mode"); // KigPaper
         }
 
     }
